lib/takeaway.rb

```diff
@@ -0,0 +1,53 @@
+require_relative 'menu'
+require_relative 'order'
+
+class TakeAway 
+
+  
+
+  def initialize 
+    @menu = Menu.new
+    @order = Order.new
+    @finish = false
+  end 
+
+  def show_menu
+    @menu.print_menu
+  end
+
+  def add(input, qua = 1)
+    return puts "I'm afraid we don't have any #{input}" unless valid?(input)
+    return puts "Order already complete, unable to add item" if finish?
+    update_total(input)
+    add_order(input, qua)
+  end
+  
+  def valid?(input)
+    @menu.items.has_key?(input.to_sym)
+  end
+
+  def update_total(input)
```

Comment:
Do you think its up to the restaurant to update the order's total or should that behaviour belong to the order?